diff --git a/Internet/TFTP/tftp.h b/Internet/TFTP/tftp.h
index b8f3b4d..5e7c804 100644
--- a/Internet/TFTP/tftp.h
+++ b/Internet/TFTP/tftp.h
@@ -12,11 +12,12 @@ extern "C" {
 #endif
 
 #include <stdint.h>
+#include <stdio.h>
 
 #define F_APP_TFTP
 #define __TFTP_DEBUG__
 
-#define F_STORAGE // If your target support a storage, you have to activate this feature and implement.
+// #define F_STORAGE	// If your target support a storage, you have to activate this feature and implement.
 
 #define SOCK_TFTP		1
 
